### PR TITLE
Processing Data in Models

### PR DESCRIPTION
**Pull request by DarkxDev**
- Create model classes for all entities as shown in the ERD Diagram.
- Set up associations between models.
- Models include some additional methods.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (franclobo) [inline app/models/user.rb:4]: - [ ] Similar to the post likes and comments are generated by an author, so please 🙏🏽 add `foreign_key: :author_id` for comments, and likes.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  def last_three_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def increment_post_counter
+    update(post_counter: post_counter + 1)
+  end
```
Comment (franclobo) [inline app/models/user.rb:12]: - **[OPTIONAL]** We could also use the increment method 👇🏽
 
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

File: app/models/post.rb
Comment (franclobo) [inline app/models/post.rb:1]: - [ ] As you define the `increment_post_counter` method you should define the same method in the post model so. We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private 👇🏽 

```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def increment_comment_counter
+    update(comments_counter: comments_counter + 1)
+  end
+
+  def increment_like_counter
+    update(likes_counter: likes_counter + 1)
+  end
```
Comment (franclobo) [inline app/models/post.rb:16]: - [ ] Please 🙏🏽 remove these lines and add in the correspondent model, I mean the `increment_coment_counter` method should be in the comment model, and so on.

- **[OPTIONAL]** As suggested for the `increment_post_counter` we can use the increment method to define them. 👆🏽

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @DarkxDev, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Correct models ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @DarkxDev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉




## Highlights

- [ ] No linter errors ✔️
- [ ] Descriptive PR ✔️
- [ ] Correct Git flow ✔️
- [ ] Correct models ✔️


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**